core/src/com/unciv/logic/civilization/QuestManager.kt

```diff
@@ -320,12 +374,56 @@ class QuestManager {
     private fun getScoreForQuest(assignedQuest: AssignedQuest): Int {
         val assignee = civInfo.gameInfo.getCivilization(assignedQuest.assignee)
         return when (assignedQuest.questName) {
+            "Contest Culture" -> assignee.policies.totalAccumulatedCulture - assignedQuest.data1.toInt()
             "Contest Techs" -> assignee.tech.getNumberOfTechsResearched() - assignedQuest.data1.toInt()
             else -> 0
         }
     }
 
     //region get-quest-target
+    /**
+     * Returns a random [TileInfo] containing a Barbarian encampment within 8 tiles of [civInfo]
+     * to be destroyed
+     */
+    private fun getBarbarianEncampmentForQuest(challenger: CivilizationInfo): TileInfo? {
+        val encampments = civInfo.getCapital().getCenterTile().getTilesInDistance(8)
+                .filter { it.hasBarbarianEncampment() }.toList()
+
+        if (encampments.isNotEmpty())
+            return encampments.random()
+
+        return null
+    }
+
+    /**
+     * Returns a random resource to be connected to the [challenger]'s trade route as a quest.
+     * The resource must be a [ResourceType.Luxury] or [ResourceType.Strategic], must not be owned
+     * by the [civInfo] and the [challenger], and must be viewable by the [challenger];
+     * if none exists, it returns null.
+     */
+    private fun getResourceForQuest(challenger: CivilizationInfo): TileResource? {
+        val viewableResources = challenger.getViewableResources()
+
+        val ownedByCityStateResources = civInfo.detailedCivResources.map { it.resource }
+                .filter { it.resourceType != ResourceType.Bonus }
+
+        val ownedByMajorResources = challenger.detailedCivResources.map { it.resource }
+                .filter { it.resourceType != ResourceType.Bonus }
+
+        val notOwnedResources = civInfo.gameInfo.ruleSet.tileResources.values
+                .filter {
+                    it.resourceType != ResourceType.Bonus &&
+                            viewableResources.contains(it) &&
```

Comment:
Since you're using viewableResources as one of the conditions, you don't need to go to the ruleset. 
viewableResources.filter { it.resourceType != ResourceType.Bonus
&& !ownedByCityStateResources.contains(it) 
&&!ownedByMajorResources.contains(it) }